graphbolt/include/graphbolt/rowwise_pick.h

```diff
@@ -0,0 +1,184 @@
+/**
+ *  Copyright (c) 2023 by Contributors
+ * @file graphbolt/rowwise_pick.h
+ * @brief Contains the function definition for RowWisePickPerEtype.
+ */
+
+#ifndef GRAPHBOLT_ROWWISE_PICK_H_
+#define GRAPHBOLT_ROWWISE_PICK_H_
+
+#include <graphbolt/csc_sampling_graph.h>
+#include <graphbolt/utils.h>
+
+using namespace graphbolt::utils;
+
+namespace graphbolt {
+namespace sampling {
+
+using CSCPtr = CSCSamplingGraph*;
+using RangePickFn = std::function<torch::Tensor(
+    int64_t start, int64_t end, int64_t num_samples)>;
+using TensorList = std::vector<torch::Tensor>;
+static constexpr int kDefaultPickGrainSize = 2;
+
+/**
+ * @brief Performs row-wise picking based on the given parameters.
+ *
+ * @param graph The pointer to a csc sampling graph.
+ * @param rows The tensor containing the row indices.
+ * @param num_picks The tensor containing the number of picks per edge type.
+ * @param probs Optional tensor containing probabilities for picking.
+ * @param return_eids Boolean indicating if edge IDs need to be returned. The
+ * last TensorList in the tuple is this value when required.
+ * @param replace Boolean indicating if picking is done with replacement.
+ * @param pick_fn The function used for picking.
+ * @return A tuple containing the picked rows, picked columns, picked etypes,
+ * and picked edge IDs (if required).
+ */
+std::tuple<torch::Tensor, torch::Tensor, torch::Tensor, torch::Tensor>
+RowWisePickPerEtype(
+    const CSCPtr graph, const torch::Tensor& rows,
+    const torch::Tensor& num_picks, const torch::optional<torch::Tensor>& probs,
+    bool return_eids, bool replace, RangePickFn pick_fn) {
```

Comment:
move the implementation to rowwise_pick.cc

Reply:
Sure, let me fix.